src/modules/hip/kernel/gaussian_filter.hpp

```diff
@@ -0,0 +1,2235 @@
+    dst_f8->f1[1] = fmaf(src_f1, filter[8], dst_f8->f1[1]);
+    dst_f8->f1[2] = fmaf(src_f1, filter[7], dst_f8->f1[2]);
+    dst_f8->f1[3] = fmaf(src_f1, filter[6], dst_f8->f1[3]);
+    dst_f8->f1[4] = fmaf(src_f1, filter[5], dst_f8->f1[4]);
+    dst_f8->f1[5] = fmaf(src_f1, filter[4], dst_f8->f1[5]);
+    dst_f8->f1[6] = fmaf(src_f1, filter[3], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter[2], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack2(src_ui4.z);
+    dst_f8->f1[2] = fmaf(src_f1, filter[8], dst_f8->f1[2]);
+    dst_f8->f1[3] = fmaf(src_f1, filter[7], dst_f8->f1[3]);
+    dst_f8->f1[4] = fmaf(src_f1, filter[6], dst_f8->f1[4]);
+    dst_f8->f1[5] = fmaf(src_f1, filter[5], dst_f8->f1[5]);
+    dst_f8->f1[6] = fmaf(src_f1, filter[4], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter[3], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack3(src_ui4.z);
+    dst_f8->f1[3] = fmaf(src_f1, filter[8], dst_f8->f1[3]);
+    dst_f8->f1[4] = fmaf(src_f1, filter[7], dst_f8->f1[4]);
+    dst_f8->f1[5] = fmaf(src_f1, filter[6], dst_f8->f1[5]);
+    dst_f8->f1[6] = fmaf(src_f1, filter[5], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter[4], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack0(src_ui4.w);
+    dst_f8->f1[4] = fmaf(src_f1, filter[8], dst_f8->f1[4]);
+    dst_f8->f1[5] = fmaf(src_f1, filter[7], dst_f8->f1[5]);
+    dst_f8->f1[6] = fmaf(src_f1, filter[6], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter[5], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack1(src_ui4.w);
+    dst_f8->f1[5] = fmaf(src_f1, filter[8], dst_f8->f1[5]);
+    dst_f8->f1[6] = fmaf(src_f1, filter[7], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter[6], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack2(src_ui4.w);
+    dst_f8->f1[6] = fmaf(src_f1, filter[8], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter[7], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack3(src_ui4.w);
+    dst_f8->f1[7] = fmaf(src_f1, filter[8], dst_f8->f1[7]);
+}
+
+// -------------------- Set 1 - PKD3->PKD3 for T = U8/F32/F16/I8 --------------------
+
+// kernelSize = 3
+template <typename T>
+__global__ void gaussian_filter_3x3_pkd_tensor(T *srcPtr,
+                                               uint2 srcStridesNH,
+                                               T *dstPtr,
+                                               uint2 dstStridesNH,
+                                               uint padLength,
+                                               uint2 tileSize,
+                                               RpptROIPtr roiTensorPtrSrc,
+                                               d_float9 *filterTensor)
+{
+    int hipThreadIdx_x8 = hipThreadIdx_x << 3;
+    int id_x_o = (hipBlockIdx_x * tileSize.x * 8) + hipThreadIdx_x8;
+    int id_y_o = hipBlockIdx_y * tileSize.y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    int id_x_i = id_x_o - padLength;
+    int id_y_i = id_y_o - padLength;
+    d_float24 sum_f24;
```

Comment:
probably better to have array of float4 locally instead of defining a new type here. I see a lot of cutom data types definintions which is not necessary

Reply:
Anything above 24 has been removed  now.
24 is used too ubiquitously in the tensor code for one shot global reads and writes of 8-element vectorized HIP kernels (8 elements * 3 for RGB = 24 element global loads/stores) that gives performance benefits.